### PR TITLE
[17.0] [MIG] sale_order_revision: migration to 17.0

### PR DESCRIPTION
**Pull request by siemenv**

### CODE REVIEW

**Comment by siemenv** [comment]
Pipelines are failing because of "base_revision" dependency which is also in pull request.

**Comment by rousseldenis** [comment]
@siemenv Why have you closed the depending PR ?

**Comment by siemenv** [comment]
> @siemenv Why have you closed the depending PR ?

I had to rebuild the branch, I fixed the link to the depending PR.

File: sale_order_revision/view/sale_order.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
                     type="object"
                     class="oe_stat_button"
                     icon="fa-file-archive-o"
-                    attrs="{'invisible': ['|', ('has_old_revisions', '=', False),('revision_count', '=', 0)]}"
+                    invisible="has_old_revisions == False or revision_count == 0"
```
Comment (CRogos) [inline sale_order_revision/view/sale_order.xml]: Could also be written as 
`invisible="not has_old_revisions or revision_count == 0"`

File: sale_order_revision/view/sale_order.xml [hunk 2]
```diff
@@ -42,7 +42,7 @@
             <field name="client_order_ref" position="after">
                 <field
                     name="current_revision_id"
-                    attrs="{'invisible': [('current_revision_id', '=', False)]}"
+                    invisible="current_revision_id == False"
```
Comment (CRogos) [inline sale_order_revision/view/sale_order.xml]: `invisible="not current_revision_id"`

**Review by CRogos — COMMENTED** [review]
Tested and LGTM

Could you also cleanup the commit history?
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

**Review by MohamedOsman7 — COMMENTED** [review]
Functionality & Code LGTM (improvements suggested by @CRogos have to be made)

**Comment by siemenv** [comment]
My squashed commits are still marked as "OCA-git-bot" or "OCA-transbot", is that normal?

**Comment by CRogos** [comment]
No this shouldn't be if you follow the instructions of the link above.

git config --global --add rebase.instructionFormat "(%an <%ae>) %s"v
git rebase -i origin/17.0

replace pick by f where needed to combine and commit 

git push -f

Are there already less commits than in the initial PR or is it the same number (nothing changed)? If it is still the same, you can retry with the commands above.

**Comment by siemenv** [comment]
@CRogos thanks for the eplanation, I did a reset and started over, I think the squashing is correct now. Can you confirm please?

**Review by CRogos — APPROVED** [review]

**Review by MohamedOsman7 — APPROVED** [review]

**Comment by rousseldenis** [comment]
@siemenv Could you remove dependency commit ?

**Comment by siemenv** [comment]
> @siemenv Could you remove dependency commit ?

Where do I have to link the odoo-test-helper requirement?

**Comment by CRogos** [comment]
@siemenv you can add this file.

https://github.com/OCA/sale-workflow/blob/16.0/test-requirements.txt

**Comment by CRogos** [comment]
@siemenv  are you still working on this?

**Comment by CRogos** [comment]
@rousseldenis can you merge?

File: sale_order_revision/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    def _prepare_revision_data(self, new_revision):
+        vals = super()._prepare_revision_data(new_revision)
+        vals.update({"state": "cancel"})
+        return vals
+
+    def action_view_revisions(self):
+        self.ensure_one()
+        action = self.env.ref("sale.action_orders")
+        result = action.read()[0]
```
Comment (rousseldenis) [inline sale_order_revision/models/sale_order.py]: You should use _for_xml_id() instead.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
One remark

**Review by FrancoMaxime — COMMENTED** [review]
Hi @siemenv  Thanks for your PR.

I agree with @rousseldenis  comment.

**Comment by CRogos** [comment]
@siemenv are you still working on this?

**Comment by CRogos** [comment]
@rousseldenis could you update your review and merge?

**Comment by CRogos** [comment]
@rousseldenis kind reminder.

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**